tests/endpoints/events/biosero/test_destination_created.py

```diff
@@ -0,0 +1,162 @@
+from http import HTTPStatus
+from unittest.mock import patch, MagicMock
+from lighthouse.helpers.mongo import get_event_with_uuid
+from lighthouse.constants.fields import FIELD_EVENT_ERRORS
+
+
+import pytest
+from uuid import uuid4
+
+CACHE = {}
+
+
+def int_to_uuid(value: int) -> str:
+    if value not in CACHE:
+        CACHE[value] = str(uuid4())
+    return CACHE[value]
+
+
+# Event source partially completed
+
+
+def test_post_destination_created_missing_barcode(app, client, biosero_auth_headers, clear_events_when_finish):
+    with app.app_context():
+        response = client.post(
+            "/events",
+            data={
+                "automation_system_run_id": 123,
+                "event_type": "lh_biosero_cp_destination_created",
+                "user_id": "user1",
+                "robot": "roboto",
+            },
+            headers=biosero_auth_headers,
+        )
+
+        assert response.status_code == HTTPStatus.UNPROCESSABLE_ENTITY
+
+
+@pytest.mark.parametrize(
+    "baracoda_mock_responses",
+    [
+        {
+            "TC1": {"barcodes_group": {"id": 1, "barcodes": ["COGUK1", "COGUK2"]}},
+        }
+    ],
+)
+@pytest.mark.parametrize("run_id", [3])
+@pytest.mark.parametrize("source_barcode", ["plate_123"])
+@pytest.mark.parametrize("destination_barcode", ["HT-1234"])
+def test_post_event_partially_completed(
```

Comment:
😱

Reply:
This is a tough one to test!

Reply:
Need some rethinking on the params to keep it tidier. Some params that are in use on call (not for data) could be merged, but I didn't want to refactor anymore until I had a better plan.